soroban-test-wasms/wasm-workspace/linear_memory/src/lib.rs

```diff
@@ -27,4 +28,28 @@ impl Contract {
         }
         Bytes::from_slice(&e, &buf)
     }
+
+    // Bounce a vector of vals off the host, successfully
+    pub fn vec_mem_ok(e: Env) -> Result<(), Error> {
+        let in_buf: [Val; 3] = [Val::from(1), Val::from(2), Val::from(3)];
+        let mut out_buf: [Val; 3] = [Val::from(0); 3];
+        let vec: VecObject = e.vec_new_from_slice(&in_buf)?;
+        e.vec_unpack_to_slice(vec, &mut out_buf)?;
+        assert!(in_buf[0].shallow_eq(&out_buf[0]));
+        assert!(in_buf[1].shallow_eq(&out_buf[1]));
+        assert!(in_buf[2].shallow_eq(&out_buf[2]));
+        Ok(())
+    }
+
+    // Same but with a length mismatch
+    pub fn vec_mem_bad(e: Env) -> Result<(), Error> {
+        let in_buf: [Val; 3] = [Val::from(1), Val::from(2), Val::from(3)];
+        let mut out_buf: [Val; 2] = [Val::from(0); 2];
+        let vec: VecObject = e.vec_new_from_slice(&in_buf)?;
+        e.vec_unpack_to_slice(vec, &mut out_buf)?;
+        // Should never get to these lines.
```

Comment:
This is confusing, why not assert that the call above is error and remove the statements below?

Reply:
fair, fixed